src/main/java/org/commcare/core/network/CommCareNetworkServiceGenerator.java

```diff
@@ -0,0 +1,79 @@
+package org.commcare.core.network;
+
+
+import org.commcare.util.LogTypes;
+import org.javarosa.core.services.Logger;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.concurrent.TimeUnit;
+
+import javax.net.ssl.SSLSocketFactory;
+import javax.net.ssl.X509TrustManager;
+
+import okhttp3.ConnectionSpec;
+import okhttp3.HttpUrl;
+import okhttp3.Interceptor;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import okhttp3.logging.HttpLoggingInterceptor;
+import retrofit2.Retrofit;
+
+/**
+ * Provides an instance of CommCareNetworkService.
+ * We have declared everything static in this class as we want to use the same objects (OkHttpClient, Retrofit, …) throughout the app
+ * to just open one socket connection that handles all the request and responses.
+ */
+
+public class CommCareNetworkServiceGenerator {
+
+    // Retrofit needs a base url to generate an instance but since our apis are fully dynamic it's not getting used.
+    private static final String BASE_URL = "https://www.commcarehq.org/";
```

Comment:
I'm not sure I understand that line, can you explain what you mean by that?

CommCare mobile is in no way tied to our server, so in general I'd push back on any part of the code containing explicit HTTP references to our current production environment

Reply:
I meant the same thing when quoting `since our apis are fully dynamic it's not getting used`.  If you see the Retrofit interface [here](https://github.com/dimagi/commcare-core/blob/retrofit/src/main/java/org/commcare/core/network/CommCareNetworkService.java) we are always passing the complete urls when making a request. So base_url doesn't end up getting used and acts just as a placeholder for building Retrofit instance.(Retofit needs a non null non empty properly formatted base url for creating an instance)

Reply:
If that's the case I think we should replace the `https://www.commcarehq.org/` reference with a reference to something which would never work, like `http://example.url`. If someone misuses or misconfigures the libraries with this base URL I'm worried about the possibility that they write code that *accidentally* works because the url was `www.commcarehq.org` as is their test data, since that will introduce incredibly hard to understand errors when someone moves to a non-production URL and the code is failing there.